### PR TITLE
Add a test charm for testing publish action

### PR DESCRIPTION
**Pull request by weiiwang01**

### CODE REVIEW

File: tests/workflows/integration/test-upload-charm/config.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+options: {}
```
Comment (arturo-seijas) [inline tests/workflows/integration/test-upload-charm/config.yaml:1]: Missing blank line

File: tests/workflows/integration/test-upload-charm/test.Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM ubuntu:latest
```
Comment (arturo-seijas) [inline tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: Missing blank line

Comment (arturo-seijas) [inline tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: Can we reuse the already existing OCI image to make sure eveything is connected as expected?

File: tests/workflows/integration/test-upload-charm/config.yaml (hunk 1 above)
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/config.yaml:1]: Updated, thanks!

File: tests/workflows/integration/test-upload-charm/test.Dockerfile (hunk 2 above)
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: Do you mean using a pre-existed dockerfile in the test dummy charm?

Comment (arturo-seijas) [reply tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: Yes, if you think is worth. It is just a suggestion

File: tests/workflows/integration/test-upload-charm/.trivyignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Ignore WordPress known vulnerabilities for version 5.5.9 (last mayor version 5)
+CVE-2022-29248
+CVE-2022-31042
+CVE-2022-31043
+CVE-2022-31090
+CVE-2022-31091
+CVE-2022-24775
```
Comment (arturo-seijas) [inline tests/workflows/integration/test-upload-charm/.trivyignore:7]: Bank line here

File: tests/workflows/integration/test-upload-charm/trivy.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-exit-code: 0
-timeout: 20m
-scan:
-  offline-scan: true
+disable-rules:
+# Ignore these rules
+# ssl-cert lib ships with a certificate and it's identified by trivy as a vulnerability
+  - private-key
```
Comment (arturo-seijas) [inline tests/workflows/integration/test-upload-charm/trivy.yaml:4]: Same here

File: tests/workflows/integration/test-upload-charm/test.Dockerfile [hunk 5]
```diff
@@ -1 +1,137 @@
-FROM ubuntu:latest
\ No newline at end of file
+FROM ubuntu:20.04
```
[the remark this answers is not in the document]
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: It takes about 5 mins to build and check, this is one of the fast docker build.

File: tests/workflows/integration/test-upload-charm/trivy.yaml (hunk 4 above)
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/trivy.yaml]: Updated, thanks.

File: tests/workflows/integration/test-upload-charm/.trivyignore (hunk 3 above)
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/.trivyignore]: Updated, thanks!

File: tests/workflows/integration/test-upload-charm/test.Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply tests/workflows/integration/test-upload-charm/test.Dockerfile:1]: Maybe it's worth removing some content, such as wp plugins or themes. The simpler, the better. Also, can we build on jammy instead?

[the remark this answers is not in the document]
Comment (weiiwang01) [reply tests/workflows/integration/test-upload-charm/test.Dockerfile]: Okay, I am just going to keep the apt install part of the dockerfile.

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by gtrkiller — APPROVED** [review]
LGTM

**Pull request merged**